### PR TITLE
fix: mobile registration flow

### LINKED ISSUE #763: Mobile registration flow
## Some mobile improvements

- [x] 33% of the time the wallet connection does not work from the register flow when we try to connect using Argent Mobile (works for Braavos mobile)
- [x] The information button (i) beside the subscription ticker box (and all the others) don't work
- [x]  Starknetkit: when the user is on mobile, better to hide the Argent (mobile) connector as this would open WC on desktop but does not work from mobile. Also, Argent X connector can be renamed Argent so users are not confused (only do that on mobile)

### PR DESCRIPTION
**Pull request by Marchand-Nicolas**
close: #763

### CODE REVIEW

**Review by fricoben — APPROVED** [review]
I can not test everything on Argent Mobile but it seems to work just fine ! I'll let @irisdv review the code :)

Lgtm

File: components/UI/navbar.tsx [hunk 1]
```diff
@@ -90,9 +90,16 @@ const Navbar: FunctionComponent = () => {
...
   const connectWallet = async (connector: Connector) => {
-    await connectAsync({ connector });
-    localStorage.setItem("SID-connectedWallet", connector.id);
-    localStorage.setItem("SID-lastUsedConnector", connector.id);
+    try {
+      await connectAsync({ connector });
+      localStorage.setItem("SID-connectedWallet", connector.id);
+      localStorage.setItem("SID-lastUsedConnector", connector.id);
+    } catch (e) {
+      // Restart the connection if there is an error except if the user has rejected the connection
```
Comment (irisdv) [inline components/UI/navbar.tsx:98]: Why do we need to restart the connection ? It should not fail if the connector is `available`

**Review by irisdv — CHANGES_REQUESTED** [review]

File: components/UI/navbar.tsx (hunk 1 above)
Comment (Marchand-Nicolas) [reply components/UI/navbar.tsx:98]: It throws this weird error sometimes (33% of the time), even with autoconnect which excplictly checks if it's available. Trying again to connect solves this issue (which I guess I a bug on starknetkit or starknetreact or is a problem of incompatibility between lib versions).
image

image

Comment (irisdv) [reply components/UI/navbar.tsx:98]: Ah ok so maybe it takes too much time to inject itself on mobile. Let's leave it like this for now then, and hopefully we'll be able to remove it at some point.

**Review by irisdv — APPROVED** [review]
lgtm

**Pull request merged**